src/game/client/draw/drawable.cpp

```diff
@@ -175,3 +212,672 @@ void Drawable::Clear_And_Set_Model_Condition_Flags(
+
+            if (Is_KindOf(KINDOF_STRUCTURE) || Is_KindOf(KINDOF_HUGE_VEHICLE)) {
+                scale = 1.0f;
+            } else if (Is_KindOf(KINDOF_VEHICLE)) {
+                scale = 0.75f;
+            } else {
+                scale = 0.5f;
+            }
+
+            int width = Get_Icon_Info()->anims[icon]->Get_Current_Frame_Width() * scale;
+            int height = Get_Icon_Info()->anims[icon]->Get_Current_Frame_Height() * scale;
+            int x = GameMath::Fast_To_Int_Truncate(region->lo.x + (region->hi.x - region->lo.x) * 0.25f - width * 0.5f);
+            int y = region->hi.y + height * 0.25f;
+            Get_Icon_Info()->anims[icon]->Draw(x, y, width, height);
+        }
+    } else {
+        Kill_Icon(ICON_ENTHUSIASTIC);
+        Kill_Icon(ICON_SUBLIMINAL);
+    }
+}
+
+void Drawable::Draw_Bombed(IRegion2D const *region)
+{
+    Object *object = Get_Object();
+    unsigned int frame = g_theGameLogic->Get_Frame();
+
+    if (object->Test_Weapon_Set_Flag(WEAPONSET_CARBOMB)
+        && object->Get_Controlling_Player() == g_thePlayerList->Get_Local_Player()) {
+        if (Get_Icon_Info()->anims[ICON_CARBOMB] == nullptr) {
+            Get_Icon_Info()->anims[ICON_CARBOMB] = new Anim2D(s_animationTemplates[ICON_CARBOMB], g_theAnim2DCollection);
+        }
+
+        if (Get_Icon_Info()->anims[ICON_CARBOMB] != nullptr) {
+            int fwidth = Get_Icon_Info()->anims[ICON_CARBOMB]->Get_Current_Frame_Width();
+            int fheight = Get_Icon_Info()->anims[ICON_CARBOMB]->Get_Current_Frame_Height();
+            int width = GameMath::Fast_To_Int_Truncate((region->hi.x - region->lo.x) * 0.5f);
+            int height = GameMath::Fast_To_Int_Truncate((float)width / (float)(fwidth) * (float(fheight)));
+            int x = GameMath::Fast_To_Int_Truncate(region->lo.x * (region->hi.x - region->lo.x) * 0.5f - width * 0.5f);
+            int y = GameMath::Fast_To_Int_Truncate(region->lo.y * (region->hi.y - region->lo.y) * 0.5f) + 5;
+            Get_Icon_Info()->anims[ICON_CARBOMB]->Draw(x, y, width, height);
+            Get_Icon_Info()->timings[ICON_CARBOMB] = 0x3FFFFFFF;
+        }
+    } else {
+        Kill_Icon(ICON_CARBOMB);
+    }
+
+    static NameKeyType key_StickyBombUpdate = g_theNameKeyGenerator->Name_To_Key("StickyBombUpdate");
+    StickyBombUpdate *update = (StickyBombUpdate *)object->Find_Update_Module(key_StickyBombUpdate);
+
+    if (update && update->Get_Target_Object()) {
+        if (update->Has_Die_Frame()) {
+            if (Get_Icon_Info()->anims[ICON_BOMB_TIMED] == nullptr) {
+                Get_Icon_Info()->anims[ICON_BOMB_TIMED] =
+                    new Anim2D(s_animationTemplates[ICON_BOMB_REMOTE], g_theAnim2DCollection);
+
+                Get_Icon_Info()->anims[ICON_BOMB_REMOTE] =
+                    new Anim2D(s_animationTemplates[ICON_BOMB_TIMED], g_theAnim2DCollection);
+
+                float f1 = GameMath::Fast_To_Int_Ceil((update->Get_Die_Frame() - frame) * (1.0f / 30.0f));
```

Comment:
Is this meant to be `float` for sure? It is used in `short` or `integer` context.

Reply:
fixed.